sql/core/src/main/scala/org/apache/spark/sql/DataFrameStatFunctions.scala

```diff
@@ -37,6 +37,16 @@ import org.apache.spark.util.sketch.{BloomFilter, CountMinSketch}
 final class DataFrameStatFunctions private[sql](df: DataFrame) {
 
   /**
+   * Calculate the approximate quantile of numerical column of a DataFrame.
+   * @param col the name of the column
+   * @param quantile the quantile number
+   * @return the approximate quantile
+   */
+  def approxQuantile(col: String, quantile: Double, epsilon: Double): Double = {
```

Comment:
It is hard to think of a case when a user just needs a single quantile (maybe just median?). If we want to expose public APIs, I would suggest `def approxQuantiles(col: String, quantiles: Array[Double], epsilon: Double): Array[Double]`. I use `Array[Double]` here for Java compatibility.